vertexai/multimodal-all/multimodalall.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// multimodalall shows an example of understanding as multimodal input a video having audio
+package multimodalall
+
+// [START generativeaionvertexai_gemini_all_modalities]
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"path/filepath"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// generateMultimodalContent generates a response into w, based upon the prompt
+// and video provided.
+// video and image are a Google Cloud Storage paths starting with "gs://"
+func generateMultimodalContent(w io.Writer, prompt, video, image, projectID, location, modelName string) error {
```

Comment:
With all these string arguments, it's easy for callers to get confused. I'd define and pass a struct in this case.

Reply:
Or is it possible to have this take a model and just call GenerateModel, and to the client and model creation elsewhere?

Reply:
Let's try with a struct.

All the options here are compromises between sometimes conflicting goals...